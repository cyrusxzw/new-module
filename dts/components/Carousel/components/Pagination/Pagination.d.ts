import { FC } from 'react';
import type { PaginationProps } from './Pagination.types';
declare const Pagination: FC<PaginationProps>;
export { Pagination };
