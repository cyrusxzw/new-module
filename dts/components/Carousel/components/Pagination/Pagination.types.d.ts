import type { Themes } from '~/types';
declare type PaginationProps = {
  dots?: any[];
  hasFlushPagination?: boolean;
  progressIndex?: number;
  theme?: Themes;
};
export type { PaginationProps };
