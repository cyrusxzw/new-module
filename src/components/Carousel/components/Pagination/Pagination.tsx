import React from 'react';
import cx from 'classnames';
import type { PaginationType } from './Pagination.types';
import styles from './Pagination.module.css';

const Pagination: PaginationType = ({
  dots = [],
  hasFlushPagination = false,
  progressIndex = 0,
  theme = 'dark',
}) => {
  if (!dots.length) return null;

  const dotsLength = dots.length;
  const width = `${100 / dotsLength}%`;
  const positionLeft = `${(100 / dotsLength) * progressIndex}%`;
  const listClassSet = cx(
    styles.list,
    { [styles.flush]: hasFlushPagination },
    styles[theme],
  );

  return (
    <div className={styles.base} data-testid="data-testid-Carousel-Pagination">
      <div className={listClassSet}>
        <div className={styles.bar} style={{ width, left: positionLeft }} />
      </div>
    </div>
  );
};

export { Pagination };
