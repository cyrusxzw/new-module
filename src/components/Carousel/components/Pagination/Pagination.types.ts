import type { ComponentWithoutChildren, Themes } from '~/types';

type PaginationProps = {
  dots?: any[];
  hasFlushPagination?: boolean;
  progressIndex?: number;
  theme?: Themes;
};

type PaginationType = ComponentWithoutChildren<PaginationProps>;

export type { PaginationProps, PaginationType };
