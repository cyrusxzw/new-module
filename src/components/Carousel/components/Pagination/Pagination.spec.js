import React from 'react';
import { render, screen } from '@testing-library/react';
import { Pagination } from './Pagination';

describe('<Carousel.Pagination />', () => {
  it('should be defined', () => {
    expect(Pagination).toBeDefined();
  });

  it('should render base component correctly', () => {
    const { container } = render(<Pagination dots={[0]} />);

    expect(container).toMatchSnapshot();
  });

  it('should return null if `dots` array is empty', () => {
    render(<Pagination />);

    const child = screen.queryByTestId(/data-testid-Carousel-Pagination/i);

    expect(child).not.toBeInTheDocument();
  });
});
