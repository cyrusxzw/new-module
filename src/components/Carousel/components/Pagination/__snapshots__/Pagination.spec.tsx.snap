// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel.Pagination /> should render base component correctly 1`] = `
<div>
  <div
    class="base"
    data-testid="data-testid-Carousel-Pagination"
  >
    <div
      class="list dark"
    >
      <div
        class="bar"
        style="width: 100%; left: 0%;"
      />
    </div>
  </div>
</div>
`;
